plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineDeployUtil.java

```diff
@@ -0,0 +1,64 @@
+package com.google.cloud.tools.eclipse.appengine.deploy;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import com.google.gson.reflect.TypeToken;
+import java.lang.reflect.Type;
+import java.util.List;
+
+/**
+ * A utility class for App Engine deploy.
+ */
+// TODO: move into appengine-core-plugin
+public class AppEngineDeployUtil {
+  /**
+   * Parse the raw json output of the deployment.
+   *
+   * @return an object modeling the output of a deploy command
+   * @throws JsonParseException if unable to extract the deploy output information needed
+   */
+  public static DeployOutput parseDeployOutput(String jsonOutput) throws JsonParseException {
+    Type deployOutputType = new TypeToken<DeployOutput>() {}.getType();
+    DeployOutput deployOutput = new Gson().fromJson(jsonOutput, deployOutputType);
+    if (deployOutput == null
+        || deployOutput.versions == null || deployOutput.versions.size() != 1) {
+      throw new JsonParseException("Cannot get app version: unexpected gcloud JSON output format");
+    }
+    return deployOutput;
+  }
+  
+  /**
+   * Holds de-serialized JSON output of gcloud app deploy. Don't change the field names
+   * because Gson uses it for automatic de-serialization.
+   */
+  // TODO expand to include other Version attributes
+  public static class DeployOutput {
+    private static class Version {
+      String id;
+      String service;
+    }
+
+    List<Version> versions;
```

Comment:
[A sample JSON output from `gcloud`](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/1332/files#diff-deb22efc56a34260de771487d5df593dR16) shows that this is a list. We always have only one version, but I believe there are cases where `gcloud` may return multiple versions, e.g., when you deploy multiple apps with a single `gcloud` invocation.